### PR TITLE
Expose operator manifests through factory

### PR DESCRIPTION
**Pull request by ironcladlou**
Incorporate the operator manifests into code generation and expose them through
the factory to support direct integration with the new OpenShift installer.

Supports https://github.com/openshift/installer/pull/331.

### CODE REVIEW

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -21,6 +22,13 @@ const (
 	DNSConfigMap          = "assets/dns/configmap.yaml"
 	DNSDaemonSet          = "assets/dns/daemonset.yaml"
 	DNSService            = "assets/dns/service.yaml"
+
+	OperatorCustomResourrceDefinition = "manifests/00-custom-resource-definition.yaml"
```
Comment (ramr) [inline pkg/manifests/manifests.go]: Typo in name - it works but ... [OperatorCustomResour]`r`[ceDefinition]

Comment (pravisankar) [inline pkg/manifests/manifests.go]: *Resource (fix typo)

File: pkg/manifests/manifests.go [hunk 2]
```diff
@@ -37,6 +45,54 @@ func NewFactory() *Factory {
 	return &Factory{}
 }
 
+func (f *Factory) OperatorCustomResourceDefinition() (*apiextensionsv1beta1.CustomResourceDefinition, error) {
+	crd, err := NewCustomResourceDefinition(MustAssetReader(OperatorCustomResourrceDefinition))
```
Comment (pravisankar) [inline pkg/manifests/manifests.go]: Same here: CustomResource

Comment (ironcladlou) [reply pkg/manifests/manifests.go]: Fixed

File: pkg/manifests/manifests.go (hunk 1 above)
Comment (ironcladlou) [reply pkg/manifests/manifests.go]: Fixed

Comment (ironcladlou) [reply pkg/manifests/manifests.go]: Fixed

**Pull request merged**